torch/nn/modules/transformer.py

```diff
@@ -238,24 +262,14 @@ def forward(
         src_key_padding_mask_for_layers = src_key_padding_mask
         why_not_sparsity_fast_path = ''
         str_first_layer = "self.layers[0]"
-        if not isinstance(first_layer, torch.nn.TransformerEncoderLayer):
-            why_not_sparsity_fast_path = f"{str_first_layer} was not TransformerEncoderLayer"
-        elif first_layer.norm_first :
-            why_not_sparsity_fast_path = f"{str_first_layer}.norm_first was True"
+        if not hasattr(self, "use_nested_tensor"):
```

Comment:
hmmm will this ever be the case? since we always set this in `__init__` and classes that inherit from this will probably call `super.__init__()`

Reply:
Unfortunately that's not the case!  When we load a model in eager mode that was saved, we get the code that is loaded with import, but the objects that were saved.

Reply:
Gotcha